standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/dataconnector/jdbc/AbstractJDBCConnectorProvider.java

```diff
@@ -245,7 +248,12 @@ protected Connection getConnection() {
   private ResultSet fetchTablesViaDBMetaData(String regex) throws SQLException {
     ResultSet rs = null;
     try {
-      rs = getConnection().getMetaData().getTables(scoped_db, null, regex, new String[] { "TABLE" });
+        if (this.type.equalsIgnoreCase(IDataConnectorProvider.ORACLE_TYPE) ||
```

Comment:
Sorry if I missed this earlier or if this was recently added, but would it make sense to move this logic into the provider ? so the abstract provider will call something like getCatalog() and getSchema() which will return whatever is appropriate. Maybe the abstract provider can provide the default and providers can override the method. or maybe just add a util method in this class (not abstract) that returns the values for all the DBs.